### ISSUE TITLE
Feature Request: Resizable left page column in page editor

### ISSUE THREAD

**Issue by v4u6h4n** [StreamController/StreamController#78]
I have long page names for a few of my pages and I can't read the full name in the page editor as the left panel can't be resized, so being able to resize that column would be handy.

**Comment by Core447** [comment]
I would like that too, the problem is the Adw component I use ([Adw.NavigationSplitView](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/class.NavigationSplitView.html)) doesn't support that. We could switch to [Gtk.Paned](https://docs.gtk.org/gtk4/class.Paned.html) but that doesn't look that nice. An alternative might be to add a config slider to the settings (ugly but would allow us to keep the current design)

**Comment by v4u6h4n** [comment]
In a kind of related topic, I was actually wondering about the UI design of the page editor. It would be nice for their to be a more intuitive integration of the page editor and the current page settings in the main UI. Maybe I'll have a look at it later today and see if I can crystalize my thoughts on it.

**Comment by v4u6h4n** [comment]
Okay, so I have some suggestions, although I understand if your not up for it, as it might involve a little bit of work.

I rarely use the `page settings tab` at the bottom of the UI as the settings it contains don't need to be accessed often, however this is a very easily accessible button in the UI. However if I want to rename a page, or delete a page, which I do more regularly, I need to open the `page editor`, search for the page, click on the page and then edit the information. I wonder if these two functions could be integrated; move the page settings found in the `page editor` into the `page settings tab`, and move the current page settings found in the `page settings tab` to the bottom of the `page settings tab`, perhaps in its own expandable menu if you want to hide them as 'advanced settings' or something like that. You can then replace the page editor button, next to the page drop down list in the main UI, with a 'create new page' button.

I think this would result in a more intuitive experience, unify settings that relate to a similar object in the same place, and also minimise the use of extra UI windows.

**Comment by Core447** [comment]
Ok, my initial thought on this was that this is kinda unintuitive if you have the same page on multiple decks and you edit properties for all of them in the deck area but still configuring it for multiple decks. However, many people suggested to see deck specific pages so this wouldn't be an issue anymore.

**Comment by v4u6h4n** [comment]
Do you mean that people might not wrap their head around the fact that the pages are deck agnostic, because they are editing the pages in a specific deck and will intuitively think that it belongs to that deck.

If that is what you mean, well, hmm... I hadn't thought about that to be honest. I suppose that might be an initial issue for  a new user who doesn't understand how it works. I was very used streamdeck-gui's approach of desk-specific pages, but it was a very quick transition when I understand the way you've done it. Which, by the way, is so much better. You could add a little image to one of the wizards pages, that visualises that you can use one page for multiple decks. Something like this, where a, b and c are pages, and 1 and 2 are streamdecks. Also sorry if the ascii is messed up, did my best lol


```
+---------+  +---------+  +---------+
|    A    |  |    B    |  |    C    |
+---------+  +---------+  +---------+
   |            |   |            |
   |     _______|   |_______     |
   |    |                    |   |
   v    v                    v   v
+-------+                  +-------+
|   1   |                  |   2   |
+-------+                  +-------+
```

**Comment by Core447** [comment]
Ok, yeah I thought it might be useful to be able to use some pages on multiple decks. 

> Which, by the way, is so much better

So you're saying we should keep doing this?

**Comment by v4u6h4n** [comment]
Yeah, I definitely prefer your approach, and I don't actually understand the purpose of having deck-specific pages, unless it's somehow just optional. If you can set the default starting page for each deck, as you can presently, don't you effectively already have deck specific paging? Maybe I'm misunderstanding something. Is it about being able to visually restrict your view of the pages to only those of one deck to simplify editing?

**Comment by Core447** [comment]
Probably, yes. And to avoid the need to add the deck name to the page names. But the current approach is more versatile.

**Comment by v4u6h4n** [comment]
Well I can understand the first one might be a bit annoying. I'm still learning the basics of python, so I'm not sure how feasible this is, or if it would add too much latency, but one solution could be to populate the page list (in the drop down box in the main gui), with a filtered list of pages that are connected to the default page of the current streamdeck.

**Comment by Core447** [comment]
To the default page, or to the selected deck? If we just use a filter, we could even add a setting to toggle this behavior.

**Comment by v4u6h4n** [comment]
Not sure if I understand what your asking, so hopefully this answers your question. 

My thinking is that if you have assigned `page 10` as the default page of `Streamdeck 2` and you select `Streamdeck 2` in the main GUI, and then, like you suggested, select the `streamdeck` filter for the dropdown list of pages, the dropdown list of pages will show only `page 10` and the tree of pages connected to `page 10` via the page change action.

I'm not familiar enough with GTK, as a user, to know what kind of conventions it uses for this kind of stuff as I mostly used KDE before Hyprland, but perhaps you could integrate a search functionality with a filter and the drop down. Maybe something like [this](https://discourse.gnome.org/t/example-of-gtk-dropdown-with-search-enabled-without-gtk-expression/12748). I guess the drop down on the right would just be a toggle button if you want to limit filtering to just `current streamdeck` and `all`, or a dropdown filter if you want to expand functionality later.

**Comment by Core447** [comment]
I actually don't know what I was asking either...

The problem with a search bar in the page selector is that searching for stuff always selects other pages. For example if you're on page `main` and also have `page1` `page2` and you're entering `p` in the search bar, the dropdown will select `page1` and load it.
If I find a way to avoid this, I can add it.

**Comment by v4u6h4n** [comment]
lol :-) 

Oh right I see what you mean, well fortunatley its not a big deal, the app as it stands is great.

**Comment by Core447** [comment]
> I would like that too, the problem is the Adw component I use ([Adw.NavigationSplitView](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/class.NavigationSplitView.html)) doesn't support that. We could switch to [Gtk.Paned](https://docs.gtk.org/gtk4/class.Paned.html) but that doesn't look that nice. An alternative might be to add a config slider to the settings (ugly but would allow us to keep the current design)

Gonna close this issue because of this - sorry
